src/app/search/search.resource.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Created by Ilya on 04/11/15.
+ */
+
+'use strict';
+
+var _$resource = new WeakMap();
```

Comment:
I use WeakMap to store the private fields. Reason behind choosing WeakMap is, the entries of WeakMap that have objects as keys are removed once the object is garbage collected.